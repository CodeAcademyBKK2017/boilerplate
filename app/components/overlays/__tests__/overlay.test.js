import Overlay from '../overlay.component';
import React from 'react';
import renderer from 'react-test-renderer';
import  'react-native';

// Note: test renderer must be required after react-native.
// import uuid from 'uuid';

import {shallow} from 'enzyme';

describe('Overlay', () => {
  
  it('renders correctly', () => {
    const tree = renderer.create(
      <Overlay />
    );
    expect(tree).toBeDefined();
  });

  it('Overlay: renders correctly', () => {
    const tree = renderer.create(<Overlay />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
