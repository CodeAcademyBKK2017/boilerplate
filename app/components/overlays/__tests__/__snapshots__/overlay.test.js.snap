// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Overlay: renders correctly 1`] = `
<View
  style={undefined}
>
  <Modal
    animationOutType="fadeOut"
    animationType="none"
    hardwareAccelerated={false}
    onClose={[Function]}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(37, 8, 10, 0.78)",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 0,
          "padding": 20,
        }
      }
      testID={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "padding": 20,
            },
            Object {
              "backgroundColor": "#eee",
            },
          ]
        }
        testID={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
  </Modal>
</View>
`;
